erpnext/accounts/doctype/purchase_invoice/purchase_invoice.py

```diff
@@ -658,7 +665,7 @@ def make_write_off_gl_entry(self, gl_entries):
 					"credit": flt(self.base_write_off_amount),
 					"credit_in_account_currency": self.base_write_off_amount \
 						if write_off_account_currency==self.company_currency else self.write_off_amount,
-					"cost_center": self.write_off_cost_center
+					"cost_center": self.cost_center or self.write_off_cost_center
```

Comment:
priority should be given to `self.write_off_cost_center`

Reply:
If user opted to tag cost center in Balance Sheet Accounts then write-off amount should also go to Cost Center selected in parent document otherwise trial balance wouldn't match.

Reply:
ok